4.0.7: test suuite uses `nose` module

`nose` is for python 2.x and should be replaced by `pytest`, `nose2` or `unittest2`
https://docs.nose2.io/en/latest/differences.html
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-gitdb-4.0.7-2.1.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-gitdb-4.0.7-2.1.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.11, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
Using --randomly-seed=1481814927
rootdir: /home/tkloczko/rpmbuild/BUILD/gitdb-4.0.7
plugins: forked-1.3.0, shutil-1.7.0, virtualenv-1.7.0, expect-1.1.0, flake8-1.0.7, timeout-1.4.2, betamax-0.8.1, freezegun-0.4.2, aspectlib-1.5.2, toolbox-0.5, rerunfailures-9.1.1, requests-mock-1.9.3, cov-2.12.1, pyfakefs-4.5.0, flaky-3.7.0, benchmark-3.4.1, xdist-2.3.0, pylama-7.7.1, datadir-1.3.1, regressions-2.2.0, cases-3.6.3, xprocess-0.18.1, black-0.3.12, checkdocs-2.7.1, anyio-3.3.0, Faker-8.11.0, asyncio-0.15.1, trio-0.7.0, httpbin-1.0.0, subtests-0.5.0, isort-2.0.0, hypothesis-6.14.6, mock-3.6.1, profiling-1.7.0, randomly-3.8.0
collected 20 items / 2 errors / 18 selected

================================================================================== ERRORS ==================================================================================
_________________________________________________________________ ERROR collecting gitdb/test/test_pack.py _________________________________________________________________
ImportError while importing test module '/home/tkloczko/rpmbuild/BUILD/gitdb-4.0.7/gitdb/test/test_pack.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib64/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
gitdb/test/test_pack.py:29: in <module>
    from nose import SkipTest
E   ModuleNotFoundError: No module named 'nose'
_______________________________________________________________ ERROR collecting gitdb/test/db/test_pack.py ________________________________________________________________
ImportError while importing test module '/home/tkloczko/rpmbuild/BUILD/gitdb-4.0.7/gitdb/test/db/test_pack.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib64/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
gitdb/test/db/test_pack.py:19: in <module>
    from nose.plugins.skip import SkipTest
E   ModuleNotFoundError: No module named 'nose'
========================================================================= short test summary info ==========================================================================
ERROR gitdb/test/test_pack.py
ERROR gitdb/test/db/test_pack.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
============================================================================ 2 errors in 0.56s =============================================================================
```

--- a/.github/workflows/pythonpackage.yml
+++ b/.github/workflows/pythonpackage.yml
@@ -36,9 +36,9 @@ jobs:
         flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
         # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
         flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
-    - name: Test with nose
+    - name: Test with pytest
       run: |
-        pip install nose
+        pip install pytest
         ulimit -n 48
         ulimit -n
-        nosetests -v
+        pytest -v

--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,5 @@
 PYTHON = python
 SETUP = $(PYTHON) setup.py
-TESTRUNNER = $(shell which nosetests)
 TESTFLAGS =
 
 all::
@@ -37,5 +36,5 @@ clean::
 	rm -f *.so
 
 coverage:: build
-	PYTHONPATH=. $(PYTHON) $(TESTRUNNER) --cover-package=gitdb --with-coverage --cover-erase --cover-inclusive gitdb
+	PYTHONPATH=. $(PYTHON) -m pytest --cov=gitdb gitdb
 

--- a/README.rst
+++ b/README.rst
@@ -30,7 +30,7 @@ If you want to go up to 20% faster, you can install gitdb-speedups with:
 REQUIREMENTS
 ============
 
-* Python  Nose - for running the tests
+* pytest - for running the tests
 
 SOURCE
 ======
@@ -45,7 +45,7 @@ Once the clone is complete, please be sure to initialize the submodules using
 
 Run the tests with
 
- nosetests
+ pytest
 
 DEVELOPMENT
 ===========

--- a/gitdb.pro.user
+++ b/gitdb.pro.user
@@ -233,8 +233,7 @@
    </valuemap>
    <valuemap key="ProjectExplorer.Target.RunConfiguration.1" type="QVariantMap">
     <valuelist key="ProjectExplorer.CustomExecutableRunConfiguration.Arguments" type="QVariantList">
-     <value type="QString">/usr/bin/nosetests</value>
-     <value type="QString">-s</value>
+     <value type="QString">/usr/bin/pytest</value>
      <value type="QString">gitdb/test/test_pack.py</value>
     </valuelist>
     <value key="ProjectExplorer.CustomExecutableRunConfiguration.BaseEnvironmentBase" type="int">2</value>

--- a/gitdb/test/db/test_pack.py
+++ b/gitdb/test/db/test_pack.py
@@ -16,7 +16,7 @@ import os
 import random
 import sys
 
-from nose.plugins.skip import SkipTest
+import pytest
 
 class TestPackDB(TestDBBase):
 
@@ -24,7 +24,7 @@ class TestPackDB(TestDBBase):
     @with_packs_rw
     def test_writing(self, path):
         if sys.platform == "win32":
-            raise SkipTest("FIXME: Currently fail on windows")
+            pytest.skip("FIXME: Currently fail on windows")
 
         pdb = PackedDB(path)
 

--- a/gitdb/test/lib.py
+++ b/gitdb/test/lib.py
@@ -65,8 +65,8 @@ def skip_on_travis_ci(func):
     @wraps(func)
     def wrapper(self, *args, **kwargs):
         if 'TRAVIS' in os.environ:
-            import nose
-            raise nose.SkipTest("Cannot run on travis-ci")
+            import pytest
+            pytest.skip("Cannot run on travis-ci")
         # end check for travis ci
         return func(self, *args, **kwargs)
     # end wrapper

--- a/gitdb/test/test_pack.py
+++ b/gitdb/test/test_pack.py
@@ -26,7 +26,7 @@ from gitdb.fun import delta_types
 from gitdb.exc import UnsupportedOperation
 from gitdb.util import to_bin_sha
 
-from nose import SkipTest
+import pytest
 
 import os
 import tempfile
@@ -246,4 +246,4 @@ class TestPack(TestBase):
     def test_pack_64(self):
         # TODO: hex-edit a pack helping us to verify that we can handle 64 byte offsets
         # of course without really needing such a huge pack
-        raise SkipTest()
+        pytest.skip('not implemented')

